koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -388,21 +388,39 @@ def check_whitelist(self, name, target_info):
         elif pkg_cfg['blocked']:
             raise koji.BuildError("package (container)  %s is blocked for tag %s" % (name, target_info['dest_tag_name']))
 
+    def has_orchestrator(self):
+        method = getattr(self.osbs(), "create_orchestrator_build", None)
+        return callable(method)
+
     def runBuilds(self, src, target_info, arches, scratch=False,
                   yum_repourls=None, branch=None, push_url=None):
 
         self.logger.debug("Spawning jobs for arches: %r" % (arches))
 
         results = []
-        for arch in arches:
-            result = self.createContainer(
-                src, target_info, arch, scratch, yum_repourls, branch, push_url)
-            results.append(result)
+
+        kwargs = dict(
+            src=src,
+            target_info=target_info,
+            scratch=scratch,
+            yum_repourls=yum_repourls,
+            branch=branch,
+            push_url=push_url
+        )
+
+        if self.has_orchestrator():
+            kwargs["arches"] = arches
+            results = [self.createContainer(**kwargs)]
+        else:
+            for arch in arches:
```

Comment:
Even though this is the existing behavior, it's not correct. This could be even more problematic as we transition over to fully support multiple architectures.

What if we simply log an error and pick the first item in `arches`? I think just ignoring the value of `arches` altogether and setting it to `'x86_64'` is also fine.